# Steps to install 'auto-sklearn' using Anaconda env on 'Ubuntu'-'Oracle VM VirtualBox'-'Windows 10'.
# python3.8 threw 'Segmentation fault' error while fitting the model, python 3.7 does not have that issue.
# Everything else is as per installation instruction documented at https://automl.github.io/auto-sklearn/master/installation.html
# Version info:
#   Ubuntu Desktop 20.04.1 LTS
#   auto-sklearn 0.8.0

conda create -n automl_v1 python=3.7
conda activate automl_v1
(automl_v1) -VirtualBox:~$ sudo apt-get install build-essential swig
(automl_v1) -VirtualBox:~$ swig -version

SWIG Version 4.0.1

Compiled with g++ [x86_64-pc-linux-gnu]

Configured options: +pcre

Please see http://www.swig.org for reporting bugs and further information

(automl_v1) -VirtualBox:~$ conda install gxx_linux-64 gcc_linux-64 swig
(automl_v1) -VirtualBox:~$ curl https://raw.githubusercontent.com/automl/auto-sklearn/master/requirements.txt | xargs -n 1 -L 1 pip install
(automl_v1) -VirtualBox:~$ pip install auto-sklearn

# After installing auto-sklearn, example.py (https://github.com/arunjoyt/automl/blob/master/example.py) was run
# which gave the below output

(automl_v1) -VirtualBox:~$ python example.py 
/home/arunjoyt/anaconda3/envs/automl_v1/lib/python3.7/site-packages/pyparsing.py:3190: FutureWarning: Possible set intersection at position 3
  self.re = re.compile(self.reString)
2020-08-21 15:28:08.230705 Start of fit
2020-08-21 16:28:06.989103 End of fit
Accuracy score 0.9911111111111112

# Tried to run the same from Jupyter Notebook, but the Notebook crashed when fit method was called.
